back-end/www/models/model_operations/answer_operations.py

```diff
@@ -0,0 +1,390 @@
+
+    Parameters
+    ----------
+    location_id : int
+        ID of the location.
+
+    Returns
+    -------
+    answers : Answer
+        The list of answers reported with the specified location.
+    """
+    answers = Answer.query.filter_by(location_id=location_id).all()
+    return answers
+
+
+def get_answers_by_user_and_location(user_id, location_id):
+    """
+    Get a list of answers by specified user id and location.
+
+    Parameters
+    ----------
+    user_id : int
+        ID of the user.
+    location_id : int
+        ID of the location.
+    Returns
+    -------
+    answers : Answer
+        The list of answers reported by the specified user and location.
+    """
+    answers = Answer.query.filter_by(user_id=user_id, location_id=location_id).all()
+
+    return answers
+
+
+def get_gold_answer_by_location(location_id):
+    """
+    Get the (first) gold answer specified by location id.
+
+    Parameters
+    ----------
+    location_id : int
+        ID of the location.
+    Returns
+    -------
+    answers : Answer
+        The gold answer.
+    """
+    answer = Answer.query.filter_by(gold_standard_status=0, location_id=location_id).first()
+    return answer
+
+
+def remove_answer(answer_id):
+    """
+    Remove an answer.
+
+    Parameters
+    ----------
+    answer_id : int
+        ID of the answer.
+
+    Raises
+    ------
+    exception : Exception
+        When no answer is found.
+    """
+    answer = get_answer_by_id(answer_id)
+
+    if answer is None:
+        raise Exception("Cannot find answer with id ", answer_id)
+
+    db.session.delete(answer)
+    db.session.commit()
+
+
+def exam_gold_standard(location_id, land_usage, expansion):
+    """
+    Check the quality of the answer in comparison with the gold standard.
+
+    Parameters
+    ----------
+    location_id : int
+        ID of the location.
+    land_usage : int
+        User's answer of judging if the land is a farm or has buildings.
+        (check the answer table in model.py for the meaning of the values)
+    expansion : int
+        User's answer of judging the construction is expanded.
+        (check the answer table in model.py for the meaning of the values)
+
+    Return
+    ------
+    int
+        Result of the checking.
+        0 means no gold standard exists.
+        1 means passing the gold standard test.
+        2 means failing the gold standard test.
+    """
+    # gold_standard_status 0 means that the answer is a gold standard
+    gold_answer = Answer.query.filter_by(gold_standard_status=0, location_id=location_id).first()
+
+    # If the gold answer doesn't exist
+    if gold_answer is None:
+        result = 0
+        return result
+
+    # If the gold answer exists, check the correctness
+    if gold_answer.land_usage == land_usage and gold_answer.expansion == expansion:
+        result = 1
+    else:
+        result = 2
+
+    return result
+
+
+def is_answer_reliable(location_id, land_usage, expansion):
```

Comment:
I need your help in explaining what this function is doing, let's chat this Wednesday during our regular meeting about this

Reply:
Of course.

And I updated the document and code (simplified) to make it more readable.